src/ape/api/providers.py

```diff
@@ -73,14 +74,26 @@ class BlockAPI(BaseInterfaceModel):
     An abstract class representing a block and its attributes.
     """
 
-    gas_data: BlockGasAPI
-    consensus_data: BlockConsensusAPI
     num_transactions: int = 0
-    hash: Optional[Any] = None
+    hash: Optional[Any] = None  # NOTE: pending block does not have a hash
     number: Optional[int] = None
-    parent_hash: Optional[Any] = None
+    parent_hash: Any = Field(
+        EMPTY_BYTES32, alias="parentHash"
+    )  # NOTE: genesis block has no parent hash
     size: int
     timestamp: int
+    gas_limit: int = Field(alias="gasLimit")
```

Comment:
I think the gas and consensus information should only be added to the models in `ape_ethereum`.

And the DBs should be ecosystem specific, right? Starknet TXNs and blocks differ a lot of Ethereum's.

You can see Starknet's blocks here: https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L963-L1024

Reply:
Here is a better link: https://starknet.io/docs/hello_starknet/cli.html#get-block

You can see blocks have `gas_price` but nothing else. No base fee. etc.
And transactions only have `max_fee`. Not priority fee etc.